pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -397,18 +410,32 @@ async def verify_attached_bugs(self, advisories: Iterable[int]):
         async with self._elliott_lock:
             await exectools.cmd_assert_async(cmd, env=self._elliott_env_vars, stdout=sys.stderr)
 
-    async def promote_all_arches(self, release_name: str, arches: List[str], previous_list: List[str], metadata: Optional[Dict], reference_releases: Dict[str, str], tag_stable: bool):
-        futures = []
-        for arch in arches:
-            futures.append(self.promote_arch(release_name, arch, previous_list, metadata, reference_releases.get(arch), tag_stable))
+    async def promote(self, release_name: str, arches: List[str], previous_list: List[str], metadata: Optional[Dict], reference_releases: Dict[str, str], tag_stable: bool):
+        tasks = []
+        if self.beta_multi:
+            self._logger.warning("[BETA] multi/heterogeneous payload will be created.")
+            tasks.append(self._promote_heterogeneous_payload(release_name, arches, previous_list, metadata, tag_stable))
+
+        if not self.beta_multi_only:
+            tasks.append(self._promote_homogeneous_payloads(release_name, arches, previous_list, metadata, reference_releases, tag_stable))
+        else:
+            self._logger.warning("BETA: Arch-specific release payloads will not be promoted because --beta-multi-only is set.")
         try:
-            release_infos = await asyncio.gather(*futures)
+            release_infos = await asyncio.gather(*tasks)
```

Comment:
hah it was bothering me how we had it "futures" instead of "tasks", thanks for fixing that

Reply:
It basically has the same meaning. I think `tasks` will be more clear to understand.